Public/Src/Cache/ContentStore/Distributed/NuCache/RocksDbContentLocationDatabase.cs

```diff
@@ -507,6 +529,162 @@ private byte[] GetKey(ShortHash hash)
+                });
+
+            // Success for this status here may actually be an error that's been returned
+            return status.Succeeded ? status.Result : new AddOrGetContentHashListResult(status.Failure.CreateException());
+        }
+
+        private bool IsContentAvailable(OperationContext context, ContentHashList contentHashList)
+        {
+            // TODO(jubayard): implement. Since this needs to pin stuff, we need it to happen in a higher layer.
+            return true;
+        }
+
+        private byte[] SerializeContentHashList(ContentHashListWithDeterminism value)
+        {
+            return SerializeWithPool(value, (instance, writer) => instance.Serialize(writer));
+        }
+        
+        /// <inheritdoc />
+        public override IReadOnlyCollection<GetSelectorResult> GetSelectors(OperationContext context, Fingerprint weakFingerprint)
+        {
+            var result = new List<GetSelectorResult>();
+
+            var status = _keyValueStore.Use(
+                store =>
+                {
+                    var key = SerializeWeakFingerprint(weakFingerprint);
+
+                    // This only works because the strong fingerprint serializes the weak fingerprint first. Hence,
+                    // we know that all keys here are strong fingerprints that match the weak fingerprint.
+                    foreach (var kvp in store.PrefixSearch(key, columnFamilyName: nameof(Columns.CaChaaS)))
+                    {
+                        var strongFingerprint = DeserializeStrongFingerprint(kvp.Key);
+                        result.Add(new GetSelectorResult(strongFingerprint.Selector));
+                    }
+                });
+
+            if (!status.Succeeded)
+            {
+                result.Add(new GetSelectorResult(status.Failure.CreateException()));
+            }
+
+            return result;
+        }
+
+        private byte[] SerializeWeakFingerprint(Fingerprint weakFingerprint)
+        {
+            return SerializeWithPool(weakFingerprint, (instance, writer) => instance.Serialize(writer));
+        }
+
+        private byte[] SerializeStrongFingerprint(StrongFingerprint strongFingerprint)
+        {
+            return SerializeWithPool(strongFingerprint, (instance, writer) => instance.Serialize(writer));
+        }
+        
+        private ContentHashListWithDeterminism DeserializeContentHashList(byte[] data)
+        {
+            return DeserializeWithPool(data, (reader) => new ContentHashListWithDeterminism(reader));
+        }
+
+        private byte[] GetCaChaaSKey(StrongFingerprint strongFingerprint)
+        {
+            return SerializeStrongFingerprint(strongFingerprint);
+        }
+
+        private StrongFingerprint DeserializeStrongFingerprint(byte[] bytes)
+        {
+            return DeserializeWithPool(bytes, (reader) => new StrongFingerprint(reader));
+        }
+
+        #endregion

```

Comment:
And I would prefer to avoid regions...

If the region is collapsed it is hard to assess the size of the file. But if they're expanded, then they're useless.

So my rule of thumb is: if you need regions, you have issues in your code.